consensus/src/epoch_manager.rs

```diff
@@ -263,8 +266,10 @@ impl EpochManager {
             .context(format!(
                 "[EpochManager] State sync to new epoch {}",
                 ledger_info
-            ))
-        // state_computer notifies reconfiguration in another channel
+            ))?;
+
+        monitor!("reconfig", self.expect_new_epoch().await);
```

Comment:
Since GitHub won't let me comment above, is the flow of how the epoch transition described in detail somewhere - e.g. receiving an EpochChangeProof from another validator -> updating storage and then state sync before processing any more messages and why we do this?  It's a little confusing.